tools/topology/topology1/sof-icl-rt711-rt1308-rt715-hdmi.m4

```diff
@@ -40,6 +40,37 @@ ifdef(`MIC_LINK',`',
 # HDMI ID calculated based on the configuraiton
 define(HDMI_BE_ID_BASE, `0')
 
+# SDW Pin as DAIs index starts from 2 on each SDW link
+define(`ALH_PIN2', `2')
+define(`ALH_PIN3', eval(ALH_PIN2) + 1)
+define(`ALH_PIN4', eval(ALH_PIN2))
+define(`ALH_PIN5', eval(ALH_PIN2))
+
+define(`ALH_JACK_OUT_NAME', `SDW'eval(UAJ_LINK)`-Playback')
+define(`ALH_JACK_IN_NAME', `SDW'eval(UAJ_LINK)`-Capture')
+define(`ALH_AMP_OUT_NAME', `SDW'eval(AMP_1_LINK)`-Playback')
+define(`ALH_MIC_IN_NAME', `SDW'eval(MIC_LINK)`-Capture')
+
+# Reorder to match multi-function codec dai index on single SDW link
+ifdef(`MFC', `
+undefine(`ALH_PIN3')
+undefine(`ALH_PIN4')
+undefine(`ALH_PIN5')
+define(`ALH_PIN3', eval(ALH_PIN2) + 2)
+define(`ALH_PIN4', eval(ALH_PIN2) + 3)
+define(`ALH_PIN5', eval(ALH_PIN2) + 1)
```

Comment:
This depends on the endpoint order. For example, rt722 is jak playback -> jack capture -> amp playback -> mic, and cs42l43 is jack playback -> mic -> jack capture -> amp. We have to use a macro to set the pin.

Reply:
@bardliao , agree, thanks, resent the PR.